### PR TITLE
Use chaining in checks + add snapshot tests

### PR DESCRIPTION
**Pull request by olivroy**
This slows it down a little bit. (tests used to take around 72 s, now around 80 s)

The commit that speaks the most of the benefit of this is this one 1cc65f573bebb57232d73ed1bdd53ca28e6b0510

Let me know if you think this is useful and I will try to wrap it up (to avoid having too many types of error checks.

Use `check_installed()` for suggested packages handling.

Edit: Sorry for the size this PR has now...

I did a follow-up to #384 in [0d9dc6c](https://github.com/rich-iannone/DiagrammeR/pull/499/commits/0d9dc6c7443129c4a38fb11e134e570d5ceeab55)

I did a follow-up to #341, #366 and #479 in [6cc8a34](https://github.com/rich-iannone/DiagrammeR/pull/499/commits/6cc8a347a3266a16467b53f06f92a5527b3ea9c7)

I replace `which(grepl())` by `grep()` in [7a2a19f](https://github.com/rich-iannone/DiagrammeR/pull/499/commits/7a2a19fc3e8533275615872addbd58bfd10f17ea)

I use `import-standalone-type-check.R` to improve value checking.

I reduced the use of pipes in some occasions.

I use cli in favour of glue (not yet finalized)

I added some more `&&` and `||` in 

I am almost done. I finalized the checking to reduce the need for `emit_error()`, `emit_warning()` and `emit_message()` shortly.

## How to review

* I highly suggest starting by reading the changes in test files.
* Check

### CODE REVIEW

**Comment by rich-iannone** [comment]
@olivroy this is phenomenally good! Thanks for doing this 👍

**Comment by olivroy** [comment]
Woo! Glad you like it. I will implement this next week or the one after in all instances. I will let you know once it is ready for review.

File: DESCRIPTION [hunk 1]
```diff
@@ -29,7 +29,7 @@ Imports:
     purrr (>= 0.3.4),
     RColorBrewer (>= 1.1-2),
     readr (>= 2.1.1),
-    rlang (>= 0.4),
+    rlang (>= 1.1.0),
```
Comment (olivroy) [inline DESCRIPTION:32]:
```suggestion
    rlang (>= 1.1.0),
    cli,
```

**Comment by olivroy** [comment]
Hi @rich-iannone,

This is ready for your review. Sorry, I got carried away and it got big. This PR aims to change nothing except error messages. Maybe you can start  by looking at the changes in tests, which are minimal and are mainly to add snapshot tests.

This PR improves the consistency of the code generally. while improving error messages.

I just squeezed in a review of how conditions are checked. It seemed to have created a small speedup on my machine!

This is not complete. Some outstanding issues could be addressed in a follow-up. Finalize the replacement of `emit_error()`, `emit_message()` with cli for example

**Review by rich-iannone — APPROVED** [review]
Lgtm!

**Comment by rich-iannone** [comment]
@olivroy thank you so much for the hard work you put into this PR! This is one of those packages where lots of people are getting value despite a fair amount of neglect on my part. This is a great step towards modernizing the package and getting it set up for further development down the line.

This is all to say that I’m really grateful!

**Comment by olivroy** [comment]
@rich-iannone yes indeed. I will finalize the transfer of error messages / conditions to rlang / cli. Let me know if you see something that is not right. I can fix it.

Could you also re-run revdeps to make sure I didn't break anything? Seems unlikely since the package is well tested and all tests passed, but just to make sure! I made huge changes after all.

**Pull request merged**